File: INF/statgrad2/16/16.cpp
#include <iostream>
#include <cmath>
using namespace std;
int f(int n){
	if(n == 0) 
		return 0;
	else if(n % 2 == 0) 
		return f(n/2);
	else if(n % 2 != 0) 
		return f(n - 1) + 1;
}
int main() {
	int count = 0;
	for(int i = 0; i < 1000000000; ++i) {
		if(f(i) == 3)
			count++;
	}
	std::cout << count;
	return 0;
}

File: skopintseva_may/16.py
count = 0
def f(n):
    if n == 0:
        return 0
    if n % 3 == 0:
        return f(n / 3) + 4
    return f(n - 1) + n
for n in range(3000):
    if f(n) == 680:
        count += 1
    
print(count)
